lib/github/merge/local.ex

```diff
@@ -0,0 +1,229 @@
+          :conflict ->
+            :conflict
+
+          _ ->
+            link_patch_batch = link_patch_batch |> Repo.preload([:patch])
+            patch = link_patch_batch.patch
+
+            git.(["fetch", "origin", patch.commit])
+
+            {_, exit_code} =
+              git.([
+                "merge",
+                patch.commit,
+                "-m",
+                "[ci skip][skip ci][skip netlify] -bors-staging-tmp-#{patch.pr_xref}"
+              ])
+
+            case exit_code do
+              0 -> status
+              _ -> :conflict
+            end
+        end
+      end)
+
+    case merge_status do
+      :conflict ->
+        :conflict
+
+      :merged ->
+        Hooks.invoke_after_merge_hook!(workdir)
+        commit_hook_changes!(workdir)
+        persist_local_to_github!("#{batch.project.staging_branch}.tmp", workdir)
+
+        local_commit_info(batch.project.name)
+    end
+  end
+
+  defp do_handle_call({:squash_merge_batch, batch, patch_links, toml}) do
+    batch = batch |> Repo.preload([:project])
+    workdir = batch.project.name
+
+    repo_conn = Project.installation_connection(batch.project.repo_xref, Repo)
+
+    # We make sure to clone the repository for each batch, rather than
+    # trying to reuse it, because hooks may change behavior in ways that could
+    # persist between batches in ways that Git can't detect, like modifying 
+    # the .git/config.
+    File.rm_rf!(workdir)
+    init_project_repo!(batch.project, cli_creds(repo_conn))
+
+    git = fn args -> System.cmd("git", args, cd: workdir) end
+
+    git.(["fetch", "origin", batch.into_branch])
+    git.(["checkout", "origin/#{batch.into_branch}"])
+
+    Hooks.invoke_before_merge_hook!(workdir)
+
+    head =
+      patch_links
+      |> Enum.reduce(batch.into_branch, fn patch_link, prev_head ->
+        patch_link = patch_link |> Repo.preload([:patch])
+        patch = patch_link.patch
+
+        {:ok, commits} = GitHub.get_pr_commits(repo_conn, patch.pr_xref)
+        {:ok, pr} = GitHub.get_pr(repo_conn, patch.pr_xref)
+
+        {token, _} = repo_conn
+        user = GitHub.get_user_by_login!(token, pr.user.login)
+
+        git.(["fetch", "origin", patch.commit])
+        {_, 0} = git.(["merge", patch.commit])
+        %{tree: tree} = local_commit_info(workdir)
+
+        # If a user doesn't have a public email address in their GH profile
+        # then get the email from the first commit to the PR
+        user_email =
+          if user.email != nil do
+            user.email
+          else
+            Enum.at(commits, 0).author_email
+          end
+
+        # Manually create a squash commit with the directory contents after the merge
+        commit_message =
+          Batcher.Message.generate_squash_commit_message(
+            pr,
+            commits,
+            user_email,
+            toml.cut_body_after
+          )
+
+        {new_head, 0} = git.(["commit-tree", tree, "-p", prev_head, "-m", commit_message])
```

Comment:
here, when doing a local squash merge, we actually *do* have to care about the commit message, as each PR will become a separate commit, and Bors will avoid adding an additional merge commit on top; these commits do eventually get seen on master. the construction here is a little subtle; we're manually constructing a commit through the CLI by taking the repo contents from `git merge` (in `tree`), but with master as the previous commit in the history (in `prev_head`).